### ISSUE TITLE
view(dtype=None) changes type

### ISSUE THREAD

**Issue by mforbes** [numpy/numpy#14875]
According to the documentation, calling `array.view(dtype=None)` should preserve the type of the array: 

* "The default, None, results in the view having the same data-type as `a`."

However, explicitly passing `dtype=None` triggers a type conversion on complex arrays.

### Reproducing code example:

```python
>>> import numpy as np
>>> A = np.eye(2, dtype=complex)
>>> A.view()
array([[1.+0.j, 0.+0.j],
       [0.+0.j, 1.+0.j]])
>>> A.view(dtype=None)                                                                                            
array([[1., 0., 0., 0.],
       [0., 0., 1., 0.]])
```
### Numpy/Python version information:

Several different versions and platforms all have the same error:

```python
>>> import sys, numpy; print(numpy.__version__, sys.version)                                                      
1.17.2 3.7.5 (default, Oct 25 2019, 10:52:18) 
[Clang 4.0.1 (tags/RELEASE_401/final)]
>>> import sys, numpy; print(numpy.__version__, sys.version)                                                      
1.16.2 3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:22:34) 
[GCC 7.3.0]
>>> import sys, numpy; print(numpy.__version__, sys.version)                                                      
('1.16.5', '2.7.16 |Anaconda custom (64-bit)| (default, Mar 14 2019, 21:00:58) \n[GCC 7.3.0]')
```

**Comment by charris** [comment]
Seems the documentation is incorrect, `dtype(None)` is an unfortunate alias for `dtype(np.float64)`.
```
In [1]: dtype(None)                                                             
Out[1]: dtype('float64')
```
I suppose that could be changed, but the current bahavior goes way back.

**Comment by charris** [comment]
And now that I check, the actual defaults at the C level are `NULL`, which can only be specified by not passing any arguments to the `view` method.

**Comment by rossbar** [comment]
It seems this was closed by #15253
